src/bin_launcher/Oni2.re

```diff
@@ -12,10 +12,14 @@ let version = () => {
 
 let passthrough = Arg.Unit(() => ());
 let passthroughFloat = Arg.Float(_ => ());
+let passthroughString = Arg.String(_ => ());
 
 let spec = [
   ("-f", Arg.Set(stayAttached), ""),
   ("--nofork", Arg.Set(stayAttached), ""),
+  ("--debug", passthrough, ""),
+  ("--log-file", passthroughString, ""),
+  ("--log-filter", passthroughString, ""),
```

Comment:
Why is this needed? Can it not just use the flags specified in `Cli'?

Reply:
Oh, good question... This isn't really documented anywhere, and should be.

Because of the way we package - `Oni2.exe` (the launcher) needs to have an extreme minimum set of dependencies. On Linux and Mac we go through some special gymnastics to bake in the dynamic libraries used:
- `OSX` - we run the `dylibbundler` tool which remaps all the dynamic-lib paths (here: https://github.com/onivim/oni2/blob/69c05a9050cca840c2fe5d082bae7233d3447566/scripts/release.js#L157)
- `Linux` - we run the `linuxdeploy` tool, which does a similar thing - moves the dynamic libs and remaps the references in `Oni2_editor` to point to them: https://github.com/onivim/oni2/blob/69c05a9050cca840c2fe5d082bae7233d3447566/scripts/linux/package-linux.sh#L26

We only run it for `Oni2_editor`, though, and not `Oni2`. `Oni_Core` has some dependencies that need that dynamic-library-packaging to work, like `textmate`, `Revery`, etc. So if the `Oni2.exe` launcher brought in `Oni_Core`, we'd get linker errors when the user tries to run it from a packaged build...

A way we could re-use modules like the `Cli` one, though, is to create another module that has no external dependencies - those would be safe to use between binaries. Something like `Oni_Common` or `Oni_Bootstrap`, with a `dune` file that has nothing but pure-ocaml dependencies.

Alternatively... we could reuse the same binary as a 'launcher' (remove building two binaries: `Oni2_editor` and `Oni2.exe`. The only reason we need the launcher, at the moment, is to detach from the Windows console... When we fix https://github.com/onivim/oni2/issues/523, the need for this split might go away completely.

Reply:
Ah, thanks for explaining! Could we not just ignore omit the ones we don't need to handle and ignore the `Arg.Bad` exceptions? I assume it'll produce an error when parsed in `Oni2_editor` anyway?

Reply:
Still a bit curious about this @bryphe

Reply:
Unless I'm mistaken, we'd still want the ones we don't want to handle here, such that they show up when you call `oni2 --help`. Unless there is a nicer way of achieveing that here?

(That is what I would say if we had some help text rather than just `""`, but I think currently none of them show up on `oni2 --help` as we haven't populated that string yet. If that is the case, we should probably add something to the strings to help.)

Reply:
There's an exception that happens on `--help` as well, so I think we can just forward that too.

Otherwise, yeah, `--help` doesn't show anything useful right now and it would be nice if it did. We do have a few options and it's starting to get hard to remember them all.